takari-builder-common/src/main/java/io/takari/builder/internal/pathmatcher/FileMatcher.java

```diff
@@ -146,36 +160,59 @@ public boolean matches(String path) {
     return true;
   }
 
-  private FileMatcher(String basedir, Matcher includesMatcher, Matcher excludesMatcher) {
-    this.basedir = basedir;
-    this.includesMatcher = includesMatcher;
-    this.excludesMatcher = excludesMatcher;
-  }
-
   public boolean matches(Path file) {
-    return matches(toAbsolutePath(file));
+    return matches(normalize0(file));
   }
 
   public boolean matches(File file) {
-    return matches(file.getAbsolutePath());
+    return matches(normalize0(file.toPath()));
   }
 
   /**
    * Given a directory, returns a map of location to FileMatcher that will optimize the lookup. The
    * key can either be a file (if it is a single path matcher) or a directory (if it is an
    * open-ended matcher). The associated matcher will still be relative to the basedir, not to the
-   * key.
-   * 
-   * @param basedir
-   * @param includes
-   * @param excludes
-   * @return
+   * key, but will only match paths that start with the key.
+   */
+  public static Map<Path, FileMatcher> createMatchers(final Path basedir,
+      Collection<String> includes, Collection<String> excludes) {
+    String basepath = normalize0(basedir);
+    if (includes == null || includes.isEmpty()) {
+      return Collections.singletonMap(basedir, createMatcher(basepath, includes, excludes));
+    }
+    return createMatchers(basepath, includes, excludes, PathNormalizer::toPath);
+  }
+
+  /**
+   * Returns a map of location to FileMatcher that will optimize the lookup. The key is a path of a
+   * file or directory in a logical filesystem that uses '/' as file separator. The associated
+   * matcher is relative to the root of the logical filesystem, not to the ley, but will only patch
+   * paths that start with the key.
    */
-  public static Map<Path, FileMatcher> subMatchers(final Path basedir, Collection<String> includes,
+  public static Map<String, FileMatcher> createMatchers(Collection<String> includes,
       Collection<String> excludes) {
+    String basepath = "";
     if (includes == null || includes.isEmpty()) {
-      return Collections.singletonMap(basedir, absoluteMatcher(basedir, includes, excludes));
+      return Collections.singletonMap(basepath, createMatcher(basepath, includes, excludes));
     }
+    return createMatchers(basepath, includes, excludes, Function.identity());
+  }
+
+  private static <T> Map<T, FileMatcher> createMatchers(String basepath,
+      Collection<String> includes, Collection<String> excludes, Function<String, T> fromString) {
+    final Matcher excludesMatcher = fromStrings(basepath, excludes, MATCH_EVERYTHING);
+    Map<T, FileMatcher> matchers = new HashMap<>();
+    newIncludesTrie(includes).subdirs().forEach((relpath, globs) -> {
+      String path = relpath != null ? basepath + "/" + relpath : basepath;
```

Comment:
Minor. Should you use PathNormalizer.SEPARATOR instead of "/" here?

Reply:
Probably, but we use hardcoded '/' is so many other places I don't think it matters. Unless we decide to use SEPARATOR everywhere.